quil-rs/src/parser/common.rs

```diff
@@ -178,7 +178,10 @@ pub(crate) fn parse_matrix<'a>(
             token!(NewLine),
             preceded(
                 token!(Indentation),
-                separated_list0(token!(Comma), parse_expression),
+                separated_list0(
+                    pair(token!(Comma), many0(token!(Indentation))),
```

Comment:
This fix might be a little too targeted, but changing how indentation is handled is a much more heavy handed fix.

The Quil spec states that indentation is exactly four spaces and can only occur at the start of a line. The new line is part of the definition of indentation, which means that indenting more than once in a single line isn't possible. It's unclear if this is intentional or not. Nothing in the Quil spec states that nested definitions are invalid, but they would require multiple levels of indentation.

Reply:
Is this only when someone wants to align the non-leftmost items within a matrix? Would it necessarily be a `Indentation` token or could it be any amount of whitespace?

Reply:
This use case is just for aligning items in the matrix. The lexer conisders a tab character or four consecutive spaces as a single indentation token. Other whitespace is ignored.

Reply:
oops - I'll unresolve since this was your thread originally